src/solvers/dg/dg.jl

```diff
@@ -246,7 +246,7 @@ end
 end
 
 
-function allocate_coefficients(mesh::TreeMesh, equations, dg::DG, cache)
+function allocate_coefficients(mesh::Union{TreeMesh, StructuredMesh}, equations, dg::DG, cache)
```

Comment:
Since this function is also valid for both DG mesh types, shouldn't this be moved to `dg_common.jl` as well?